lib/transaction/logs.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+/**
+ * Storage mechanism for logs within a transaction
+ * As logs are seen during a transaction they are added to the storage
+ * if max limit has not been met.  When a transaction ends they are flushed
+ * and added to the Log Aggregator and then cleared
+ *
+ * @param {object} agent the agent
+ */
+function Logs(agent) {
+  this.maxLimit = agent.config.event_harvest_config.harvest_limits.log_event_data
+  this.aggregator = agent.logs
+  this.storage = []
+}
+
+/**
+ * Adds a log line to storage if max limit has not been met
+ *
+ * @param {object} logLine log line to store
+ */
+Logs.prototype.add = function add(logLine) {
+  if (this.storage.length >= this.maxLimit) {
```

Comment:
Not sure the `>` of that `>=` is possible, but good catch on allowing the length to be equal to the limit, as normally used on the corresponding priority queue.

https://github.com/newrelic/node-newrelic/blob/main/lib/priority-queue.js#L37

Reply:
Hmm. I'll update this to be consistent, good catch